src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.Changed.cs

```diff
@@ -118,4 +118,128 @@ public static void FileSystemWatcher_Changed_FileInNestedDirectory()
         },
         NotifyFilters.DirectoryName | NotifyFilters.LastAccess | NotifyFilters.FileName);
     }
+
+    [Fact]
+    public static void FileSystemWatcher_Changed_FileDataChange()
+    {
+        using (var dir = Utility.CreateTestDirectory())
+        using (var watcher = new FileSystemWatcher())
+        {
+            AutoResetEvent eventOccured = Utility.WatchForEvents(watcher, WatcherChangeTypes.Changed);
+
+            using (FileStream file = File.Create(Path.Combine(dir.Path, "testfile.txt")))
+            {
+                // Attach the FSW to the existing structure
+                watcher.Path = Path.GetFullPath(dir.Path);
+                watcher.Filter = "*";
+                watcher.NotifyFilter = NotifyFilters.Attributes | NotifyFilters.CreationTime | NotifyFilters.DirectoryName | NotifyFilters.FileName | NotifyFilters.LastAccess | NotifyFilters.LastWrite | NotifyFilters.Security | NotifyFilters.Size;
+                watcher.EnableRaisingEvents = true;
+
+                // Change the nested file and verify we get the changed event
+                byte[] bt = new byte[8000];
+                file.Write(bt, 0, bt.Length);
+                file.Flush();
+            }
```

Comment:
Does the directory created by CreateTestDirectory delete itself and all contained files when 'dir' is disposed?